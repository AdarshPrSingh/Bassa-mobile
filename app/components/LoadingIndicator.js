// @flow
import React from 'react';
import {
  View,
  Dimensions,
  ViewPropTypes,
  StyleSheet,
} from 'react-native';
import Spinner from 'react-native-spinkit';
import PropTypes from 'prop-types';

const HEIGHT: number = Dimensions.get('window').height;

const TOP_NORMAL = (HEIGHT / 2) - 10;
const TOP_WITH_NAV = (HEIGHT / 2) - 65;
const TOP_WITH_NAV_AND_TAB = (HEIGHT / 3);

const calculateTop = (position, marginTop) => {
  switch (position) {
    case 'nav':
      return TOP_WITH_NAV;
    case 'nav_tabs':
      return TOP_WITH_NAV_AND_TAB;
    case 'custom':
      return marginTop;
    default:
      return TOP_NORMAL;
  }
};

const LoadingIndicator = ({
  color, isVisible, position, marginTop, size, type, style,
}) => (
    <View
      pointerEvents={'none'} style={[
        styles.loaderContainer,
        { top: calculateTop(position, marginTop) },
      ]}>
      <Spinner
        style={style}
        isVisible={isVisible}
        size={size}
        type={type}
        color={color} />
    </View>
);

LoadingIndicator.propTypes = {
  color: PropTypes.string,
  isVisible: PropTypes.bool.isRequired,
  position: PropTypes.string,
  marginTop: PropTypes.number,
  style: ViewPropTypes.style,
  type: PropTypes.string,
  size: PropTypes.number,
};

LoadingIndicator.defaultProps = {
  color: '#FFF',
  isVisible: false,
  size: 60,
  type: 'Bounce',
  style: {},
  position: '',
  marginTop: 0,
};

export default LoadingIndicator;

const styles = StyleSheet.create({
  loaderContainer: {
    position: 'absolute',
    alignSelf: 'center',
  },
});
