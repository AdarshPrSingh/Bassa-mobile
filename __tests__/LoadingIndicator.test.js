import React from 'react';
import LoadingIndicator from '../app/components/LoadingIndicator';

import renderer from 'react-test-renderer';

describe('LoadingIndicator component', () => {
    test('LoadingIndicator component renders correctly', () => {
        const tree = renderer.create(<LoadingIndicator />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});