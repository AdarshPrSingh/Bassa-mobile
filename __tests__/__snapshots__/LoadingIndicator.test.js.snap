// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator component LoadingIndicator component renders correctly 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      Object {
        "alignSelf": "center",
        "position": "absolute",
      },
      Object {
        "top": 657,
      },
    ]
  }
>
  <View />
</View>
`;
